rust/src/bin/mina-indexer.rs

```diff
@@ -198,10 +217,13 @@ pub async fn main() -> anyhow::Result<()> {
 
             init_tracing_logger(log_dir.clone(), log_level_file, log_level).await?;
 
-            // write server args to config.json
-            let path = log_dir.with_file_name("config.json");
+            // dump server config
+            let path = args
+                .config_file
+                .clone()
+                .unwrap_or(log_dir.with_file_name("config.json"));
             let args_json: ServerArgsJson = args.clone().into();
-            std::fs::write(path, serde_json::to_string_pretty(&args_json)?)?;
+            std::fs::write(&path, serde_json::to_string_pretty(&args_json)?)?;
```

Comment:
Does this line not overwrite the given config file?

Reply:
This must not be allowed. A first reason is because config files are often installed in a read-only manner. Second, if a sysadmin ever found the modification date on their config file to be other than when they modified it, then they would be astonish or suspicious that their config has been hacked. Third, there is no need to write the file if the configs have not changed. (Can one change the configs on the command line in addition to providing a config file? Often, this is possible, but if so then one does not expect the config file to change.)

Reply:
will change